openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataQuality/DataQualityWorkflow.java

```diff
@@ -0,0 +1,195 @@
+package org.openmetadata.service.apps.bundles.insights.workflows.dataQuality;
+
+import static org.openmetadata.service.apps.bundles.insights.utils.TimestampUtils.END_TIMESTAMP_KEY;
+import static org.openmetadata.service.apps.bundles.insights.utils.TimestampUtils.START_TIMESTAMP_KEY;
+import static org.openmetadata.service.workflows.searchIndex.ReindexingUtil.ENTITY_TYPE_KEY;
+import static org.openmetadata.service.workflows.searchIndex.ReindexingUtil.getTotalRequestToProcess;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.EntityTimeSeriesInterface;
+import org.openmetadata.schema.service.configuration.elasticsearch.ElasticSearchConfiguration;
+import org.openmetadata.schema.system.IndexingError;
+import org.openmetadata.schema.system.StepStats;
+import org.openmetadata.service.apps.bundles.insights.DataInsightsApp;
+import org.openmetadata.service.apps.bundles.insights.utils.TimestampUtils;
+import org.openmetadata.service.apps.bundles.insights.workflows.WorkflowStats;
+import org.openmetadata.service.exception.SearchIndexException;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.search.elasticsearch.ElasticSearchEntityTimeSeriesProcessor;
+import org.openmetadata.service.search.elasticsearch.ElasticSearchIndexSink;
+import org.openmetadata.service.search.models.IndexMapping;
+import org.openmetadata.service.search.opensearch.OpenSearchEntityTimeSeriesProcessor;
+import org.openmetadata.service.search.opensearch.OpenSearchIndexSink;
+import org.openmetadata.service.util.ResultList;
+import org.openmetadata.service.workflows.interfaces.Processor;
+import org.openmetadata.service.workflows.interfaces.Sink;
+import org.openmetadata.service.workflows.interfaces.Source;
+import org.openmetadata.service.workflows.searchIndex.PaginatedEntityTimeSeriesSource;
+
+@Slf4j
+public class DataQualityWorkflow {
+  public static final String DATA_STREAM_KEY = "DataStreamKey";
+  private final int retentionDays = 30;
```

Comment:
How do we use this value? Just want to make sure we will not delete index data that are older than 30 days. 😊

Reply:
this restricts user for putting backfill date older then 30 days, do we here want to allow user to enable backfill upto 
any date?

Reply:
Yes, we should